api/v1/utils.go

```diff
@@ -121,33 +125,87 @@ func GetWorkDirectory(aerospikeConfigSpec AerospikeConfigSpec) string {
 	return DefaultWorkDirectory
 }
 
-func getInitContainerImage(registry string) string {
+func getInitContainerImage(registry, namespace, repoAndTag string) string {
 	return fmt.Sprintf(
 		"%s/%s/%s", strings.TrimSuffix(registry, "/"),
-		strings.TrimSuffix(AerospikeInitContainerDefaultRegistryNamespace, "/"),
-		AerospikeInitContainerDefaultRepoAndTag,
+		strings.TrimSuffix(namespace, "/"),
+		repoAndTag,
 	)
 }
 
 func GetAerospikeInitContainerImage(aeroCluster *AerospikeCluster) string {
+	registry := getInitContainerImageValue(aeroCluster, AerospikeInitContainerRegistry)
+	namespace := getInitContainerImageRegistryNamespace(aeroCluster)
+	repoAndTag := getInitContainerImageValue(aeroCluster, AerospikeInitContainerNameAndTag)
+
+	return getInitContainerImage(registry, namespace, repoAndTag)
+}
+
+func getInitContainerImageRegistryNamespace(aeroCluster *AerospikeCluster) string {
 	// Given in CR
-	registry := ""
+	var namespace *string
 	if aeroCluster.Spec.PodSpec.AerospikeInitContainerSpec != nil {
-		registry = aeroCluster.Spec.PodSpec.AerospikeInitContainerSpec.ImageRegistry
+		namespace = aeroCluster.Spec.PodSpec.AerospikeInitContainerSpec.ImageRegistryNamespace
+	}
+
+	if namespace == nil {
+		// Given in EnvVar
+		envRegistryNamespace, found := os.LookupEnv(AerospikeInitContainerRegistryNamespaceEnvVar)
+		if found {
+			namespace = &envRegistryNamespace
+		}
 	}
 
-	if registry != "" {
-		return getInitContainerImage(registry)
+	if namespace == nil {
+		return AerospikeInitContainerDefaultRegistryNamespace
 	}
 
-	// Given in EnvVar
-	registry, found := os.LookupEnv(AerospikeInitContainerRegistryEnvVar)
-	if found {
-		return getInitContainerImage(registry)
+	return *namespace
+}
+
+func getInitContainerImageValue(aeroCluster *AerospikeCluster, valueType string) string {
```

Comment:
1. Instead of defining new const for valueType why not use env constants like `AEROSPIKE_KUBERNETES_INIT_REGISTRY` for differentiation
2. Can pass the default value as well as per env constant type. 

Doing above will skip 2 switch cases in this func